### PR TITLE
Feature/9669 sort collection for paired collections

### PR DESCRIPTION
**Pull request by assuntad23**
Adding new type to sort collection tool: paired lists

Fulfilling request in issue #9669

### CODE REVIEW

**Review by nsoranzo — COMMENTED** [review]
For consistency with the rest of the codebase, I'd suggest to replace all instances of `force_flush` with `flush` in `lib/galaxy/model/__init__.py` and `lib/galaxy/tools/__init__.py` .

File: lib/galaxy/model/__init__.py [hunk 1]
```diff
@@ -4537,13 +4538,14 @@ def dataset_instances(self):
...
         if element_destination:
             if self.is_collection:
                 element_object = element_object.copy(
                     destination=destination,
-                    element_destination=element_destination
+                    element_destination=element_destination,
+                    force_flush=force_flush
                 )
             else:
                 new_element_object = element_object.copy()
```
Comment (jmchilton) [inline lib/galaxy/model/__init__.py]: copy(force_flush=force_flush)

File: lib/galaxy/tools/sort_collection_list.xml [hunk 2]
```diff
@@ -22,7 +22,7 @@
         </conditional>
     </inputs>
     <outputs>
-        <collection name="output" format_source="input" type="list" label="${on_string} (sorted)" >
+        <collection name="output" format_source="input" type_source="input" label="${on_string} (sorted)" >
         </collection>
     </outputs>
     <tests>
```
Comment (jmchilton) [inline lib/galaxy/tools/sort_collection_list.xml:28]: Can you add a test case to the follow block - checkout flatten_collection.xml in the same file for some ideas about syntax and such?

Comment (assuntad23) [reply lib/galaxy/tools/sort_collection_list.xml:28]: Done.

File: lib/galaxy/model/__init__.py (hunk 1 above)
Comment (assuntad23) [reply lib/galaxy/model/__init__.py]: Done.

File: lib/galaxy/tools/sort_collection_list.xml (hunk 2 above)
Comment (assuntad23) [reply lib/galaxy/tools/sort_collection_list.xml:28]: Done.

File: lib/galaxy/tools/sort_collection_list.xml [hunk 3]
```diff
@@ -91,6 +91,47 @@
...
+              <element name="def456_forward">
+                <assert_contents>
+                  <has_text_matching expression="^This is a line of text.\n$" />
+                </assert_contents>
+              </element>
+              <element name="def456_reverse">
+                <assert_contents>
+                  <has_text_matching expression="^This is a different line of text.\n$" />
+                </assert_contents>
+              </element>
```
Comment (nsoranzo) [inline lib/galaxy/tools/sort_collection_list.xml]: Is this the expected content of the list of pairs?
I'd have expected something like:
```suggestion
                <element name="abc789">
                    <collection type="paired">
                        <element name="forward">
                            <assert_contents>
                                <has_text_matching expression="^This is a line of text.\n$" />
                            </assert_contents>
                        </element>
                        <element name="reverse">
                            <assert_contents>
                                <has_text_matching expression="^This is a different line of text.\n$" />
                            </assert_contents>
                        </element>
                    </collection>
                </element>
                <element name="def456">
                    <collection type="paired">
                        <element name="forward">
                            <assert_contents>
                                <has_text_matching expression="^This is a line of text.\n$" />
                            </assert_contents>
                        </element>
                        <element name="reverse">
                            <assert_contents>
                                <has_text_matching expression="^This is a different line of text.\n$" />
                            </assert_contents>
                        </element>
                    </collection>
                </element>
```

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Thanks @assuntad23 !

**Pull request merged**